compiler/rustc_borrowck/src/region_infer/opaque_types/mod.rs

```diff
@@ -129,17 +129,17 @@ fn nll_var_to_universal_region<'tcx>(
 /// Collect all defining uses of opaque types inside of this typeck root. This
 /// expects the hidden type to be mapped to the definition parameters of the opaque
 /// and errors if we end up with distinct hidden types.
-fn add_concrete_opaque_type<'tcx>(
+fn add_hidden_type<'tcx>(
     tcx: TyCtxt<'tcx>,
-    concrete_opaque_types: &mut ConcreteOpaqueTypes<'tcx>,
+    hidden_types: &mut HiddenTypes<'tcx>,
```

Comment:
`HiddenTypesInDefinitionScope` or sth 🤔 

I feel like the distinction between
```rust
type Tait<U> = impl Sized;
fn foo<T>() -> Tait<T> {}
```
`Tait<T> = T` in the defining scope and the resulting `Trait<U> = U` in the definition scope is really important and otherwise easy to miss, so making it easier to tell them apart would be useful.

Thinking about this, we just also just eagerly wrap the oapques with an `EarlyBinder` here after mapping them to their definition scope